addons/lunch/models/lunch_product.py

```diff
@@ -20,44 +21,51 @@ class LunchProduct(models.Model):
+    image_small = fields.Binary(
+        "Small-sized image", attachment=True,
+        help="Small-sized image of the product. It is automatically "
+             "resized as a 64x64px image, with aspect ratio preserved. "
+             "Use this field anywhere a small image is required.")
+
+    is_favorite = fields.Boolean('Has Already Been Ordered', compute='_compute_is_favorite')
+
+    @api.model
+    def search_read(self, domain=None, fields=None, offset=0, limit=None, order=None):
+        res = super(LunchProduct, self).search_read(domain, fields, offset, limit, order)
+
+        if not order and 'is_favorite' in fields:
+            res = sorted(res, key=lambda x: x['is_favorite'], reverse=True)
+
+        return res
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for values in vals_list:
+            tools.image_resize_images(values)
+        return super(LunchProduct, self).create(vals_list)
+
+    def write(self, values):
+        tools.image_resize_images(values)
+        return super(LunchProduct, self).write(values)
 
-    @api.depends('supplier')
-    def _get_available_product(self):
+    def _compute_is_favorite(self):
+        last_ordered = fields.Date.today() - relativedelta(weeks=2)
         for product in self:
-            if not product.supplier:
-                product.available = True
-            else:
-                alerts = self.env['lunch.alert'].search([
-                    ('partner_id', '=', self.supplier.id)
-                ])
-                if alerts and not any(alert.display for alert in alerts):
-                    # every alert is not available
-                    product.available = False
-                else:
-                    # no alert for the supplier or at least one is not available
-                    product.available = True
-
-    def _search_available_products(self, operator, value):
-        alerts = self.env['lunch.alert'].search([])
-        supplier_w_alerts = alerts.mapped('partner_id')
-        available_suppliers = alerts.filtered(lambda a: a.display).mapped('partner_id')
-        available_products = self.search([
-            '|',
-                ('supplier', 'not in', supplier_w_alerts.ids),
-                ('supplier', 'in', available_suppliers.ids)
-        ])
-
-        if (operator in expression.NEGATIVE_TERM_OPERATORS and value) or \
-           (operator not in expression.NEGATIVE_TERM_OPERATORS and not value):
-            # e.g. (available = False) or (available != True)
-            return [('id', 'not in', available_products.ids)]
-        else:
-            # e.g. (available = True) or (available != False)
-            return [('id', 'in', available_products.ids)]
+            product.is_favorite = bool(self.env['lunch.order.line'].search_count([('product_id', '=', product.id), ('date', '>=', last_ordered), ('user_id', '=', self.env.user.id)]))
```

Comment:
There is a way to write this with a `read_group` . Like `read_group([('product_id', 'in', self.ids), ('date', '>=', last_ordered), ('user_id', '=', self.env.user.id)], [], ['product_id'], lazy=False)`.
So, you'll get a result grouped by product and containing the '__count' key. 
Not sure, but should be tested ;)